docs/ADRs/0003-ci-cd.md

```diff
@@ -0,0 +1,30 @@
+---
+    title: 0003 Continuous Integration/Continuous Deployment (CI/CD) Tooling
+    adr:
+        author: Berry den Hartog
+        created: 28-Feb-2024
+        status: accepted
+---
+
+## Context
+
+Our development team wants to implement a CI/CD solution to streamline the build,
+testing, and deployment workflows of our software products. Currently, our codebase resides on GitHub, and we leverage
+Kubernetes as our chosen orchestration platform, managed by the DigiLab platform team.
+
+## Decision
+
+We will use the following tools for CI/CD pipeline:
+
+* Continuous Integration (CI): GitHub Actions will be employed to facilitate the automated testing of our applications.
+* Continuous Deployment (CD): We will utilize Flux for managing the deployment process of our applications. Flux reads
```

Comment:
why would we use flux for CD, could we also use Github Actions for it? Or is this blocked by us not having a verified version of Github?

Reply:
Flux pulls the repo every x minutes and makes sure the kubernetes jobs are in sync with the repository. We can not choose a different tool because this created and maintained by digilabs infra team. (other option could be ArgoCD)

We could use a github action to deploy to kubernetes but it would invalidate the use of flux. i prefer flux because it gives you the consistency that what is in the repo is actually how it is running in kubernetes. if you do it with github actions you do not know if someone changed the deployment through kubectl or any other way.

Reply:
alright that makes sense that this is restricted by the digilabs infra team and I also agree with your reasoning in the second paragraph :)